specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json

```diff
@@ -3586,6 +3590,105 @@
         }
       }
     },
+    "FunctionsDeployment": {
+      "description": "Configuration section for Azure Function app deployment.",
+      "type": "object",
+      "properties": {
+        "storage": {
+          "description": "Storage for deployed package used by the Azure Function app.",
+          "type": "object",
+          "properties": {
+            "type": {
+              "description": "Property to select Azure Storage Type. Available option: blobContainer.",
+              "type": "string"
+            },
+            "value": {
+              "description": "Property to set the URL for the selected Azure Storage Type. Example: For blobContainer, value will be like https://<storageAccountName>.blob.core.windows.net/<containerName>.",
+              "type": "string"
+            },
+            "authentication": {
+              "description": "Authentication method to access the storage for deployment.",
+              "type": "object",
+              "properties": {
+                "type": {
+                  "description": "Property to select Authentication Type to access the selected storage. Available options: SystemAssignedIdentity, UserAssignedIdentity, StorageAccountAccessKey.",
+                  "type": "string"
+                },
+                "value": {
+                  "description": "Property to set the value for the selected Authentication Type. Example: For SystemAssignedIdentity, do not set value. For UserAssignedIdentity, set resource ID of identity. For StorageAccountAccessKey, set name of the app setting that has the storage account access key.",
+                  "type": "string"
+                }
```

Comment:
I think its quite a confusing API design to create a single property 'value', then give it 3 overloaded meanings, and creates many challenges for e.g. doing good validation tooling. I would suggest instead having separate properties for every separate meaning, and explaining that each property has only this one kind of valid value, and that you are only allowed to set one of the values at a time.

Reply:
Thank you for the suggestion. I have updated the definition so that the type property uses an enum to restrict its value to one of the three specified authentication types. Each authentication type (`systemAssignedIdentity`, `userAssignedIdentityResourceId`, `storageAccountAccessKeyName`) remains conditionally required and is used based on the value of the type property. In addition, I have added the `oneOf` keyword to be used in conjunction with required and properties to ensure that the correct property is provided for each authentication type.

Reply:
One more thing, for the `SystemAssignedIdentity` authentication type, there is no need to supply any additional parameters. What are you thoughts on not including an authentication value for this type?

Reply:
Removed `SystemAssignedIdentity` authentication value given that no additional properties are needed for this authentication type.